src/state/session/agent-wrapper.ts

```diff
@@ -0,0 +1,31 @@
+import {AtpSessionEvent, BskyAgent} from '@atproto/api'
+
+/**
+ * This is a temporary class that wraps `BskyAgent` and allows us to update the session persist handler when needed,
```

Comment:
I wouldn't necessarily keep it temporary, it's nice to be able to preconfigure some things here. I might set up some other overrides for the online management.